common/Path.cc

```diff
@@ -0,0 +1,107 @@
+#include "common/Path.h"
+
+namespace sorbet {
+using namespace std;
+
+#if defined(_WIN32) || defined(_WIN64)
+constexpr char file_separator = '\\';
+#else
+constexpr char file_separator = '/';
+#endif
+
+/**
+ * A simple naive implementation of Path for platform that do not have <filesystem> yet
+ */
+
+template <> BasePath<string>::BasePath(const string_view path) : internal_path(std::string(path)) {}
+
+template <> BasePath<string> &BasePath<string>::replace_filename(const string_view filename) {
+    auto last_sep = internal_path.rfind(file_separator);
+    if (last_sep == std::string::npos)
+        internal_path = filename;
+    else
+        internal_path.replace(last_sep + 1, internal_path.size() - last_sep - 1, filename);
+
+    return *this;
+}
+
+template <> BasePath<string> &BasePath<string>::replace_extension(const string_view extension) {
+    auto last_sep = internal_path.rfind(file_separator);
+    last_sep = last_sep == string::npos ? static_cast<size_t>(-1) : last_sep;
+    auto last_dot = internal_path.find('.', last_sep + 1);
+    if (last_dot == string::npos)
+        internal_path += extension;
+    else
+        internal_path.replace(last_dot, internal_path.size() - last_dot, extension);
+
+    return *this;
+}
+
+template <> BasePath<string> &BasePath<string>::lexically_normal() {
+    // TODO
```

Comment:
????

Reply:
Ah crap I forgot about leaving this. I will need to implement that somehow. Flipping back the WIP bit.